core/src/main/scala/org/apache/spark/deploy/worker/ui/LogPage.scala

```diff
@@ -137,8 +143,16 @@ private[ui] class LogPage(parent: WorkerWebUI) extends WebUIPage("logPage") with
       return ("Error: invalid log directory " + logDirectory, 0, 0, 0)
     }
 
+    // Find a log file name
+    val fileName = if (logType.equals("out")) {
+      normalizedLogDir.listFiles.map(_.getName).filter(_.endsWith(".out"))
+        .headOption.getOrElse(logType)
```

Comment:
What the "out" file is? Is it Worker's log fille?

Reply:
Yes, all Spark deamon's log file follows that.

https://github.com/apache/spark/blob/25df5dad67610357287bef075ee755c59acdb904/sbin/spark-daemon.sh#L125

Reply:
It seems inappropriate to always peek the head as we support multiple `$instance`s on a single machine.

Reply:
Ya, that's a possible configuration, but this PR is supposed to be used with an explicitly user-given `SPARK_LOG_DIR`, @yaooqinn . In addition, in these day, Spark Standalone cluster is running on top of K8s environment. I believe we can clarify the assumption during documentation, @yaooqinn .

Reply:
To be clear, historically, spark worker log is generated outside `work` directory for security reasons. So, it means we should not expose it without human guiadance.

Reply:
I agree with you that `SPARK_LOG_DIR` for each or being isolated by Kubernetes does solve the issue. However, exporting `SPARK_LOG_DIR` system or session-wide may still pose problems. Besides standalone clusters, I am wondering the local-cluster mode I use extensively for dev might also have such an issue.

Reply:
> - `SPARK_WORKER_INSTANCES` is deprecated in Standalone mode. It's recommended to launch multiple executors in one worker and launch one worker per node instead of launching multiple workers per node and launching one executor per worker.

I found this statement. I agree w/ you that we doc this. @dongjoon-hyun

Reply:
Thank you!